api/indexer/tx_indexer.go

```diff
@@ -49,11 +55,11 @@ func WithIndexer(name string, path string) vm.Option {
 			indexer: indexer,
 		}
 
-		if err := vm.WithBlockSubscriptions(subscriptionFactory)(v); err != nil {
+		if err := vm.WithBlockSubscriptions(subscriptionFactory)(v, configBytes); err != nil {
```

Comment:
`WithBlockSubscriptions` should never actually take in `configBytes` since it should only be called to modify the VM. Same for `WithVMAPIs`.

There should be the namespaced options, which register a service and separately the functions that are not registering a service, but are only modifying the VM. For example, `WithBlockSubscriptions` should not be called directly, but to register a specific service inside of a namespaced option. There's no need for this to take in `configBytes`, so I think we can remove it from the interface.